src/NServiceBus.Persistence.AzureTable/SagaPersisters/SecondaryIndices/SecondaryIndex.cs

```diff
@@ -31,26 +32,25 @@ public SecondaryIndex(bool assumeSecondaryIndicesExist = false, bool assumeSecon
 
             var rowKey = assumeSecondaryKeyUsesANonEmptyRowKeySetToThePartitionKey ? key.PartitionKey : key.RowKey;
 
-            TableResult exec;
+            Response<SecondaryIndexTableEntity> exec;
             try
             {
-                exec = await table.ExecuteAsync(TableOperation.Retrieve<SecondaryIndexTableEntity>(key.PartitionKey, rowKey), cancellationToken)
-                    .ConfigureAwait(false);
+                exec = await table.GetEntityAsync<SecondaryIndexTableEntity>(key.PartitionKey, rowKey, null, cancellationToken).ConfigureAwait(false);
             }
-            catch (StorageException storageException)
-                when (!assumeSecondaryKeyUsesANonEmptyRowKeySetToThePartitionKey && storageException.RequestInformation.HttpStatusCode == (int)HttpStatusCode.PreconditionFailed)
+            catch (RequestFailedException requestFailedException) // TODO: verify if this holds up
+                when (!assumeSecondaryKeyUsesANonEmptyRowKeySetToThePartitionKey && requestFailedException.Status == (int)HttpStatusCode.PreconditionFailed)
             {
                 Logger.Warn(
                     $"Trying to retrieve the secondary index entry with PartitionKey = '{key.PartitionKey}' and RowKey = 'string.Empty' failed. When using the compatibility mode on Azure Cosmos DB it is strongly recommended to enable `sagaPersistence.AssumeSecondaryKeyUsesANonEmptyRowKeySetToThePartitionKey()` to avoid additional lookup costs or disable the compatibility mode entirely if not needed by calling `persistence.Compatibility().DisableSecondaryKeyLookupForSagasCorrelatedByProperties(). Falling back to query secondary index entry with PartitionKey = '{key.PartitionKey}' and RowKey = '{key.PartitionKey}'",
-                    storageException);
+                    requestFailedException);
 
-                exec = await table.ExecuteAsync(TableOperation.Retrieve<SecondaryIndexTableEntity>(key.PartitionKey, key.PartitionKey), cancellationToken)
-                    .ConfigureAwait(false);
+                exec = await table.GetEntityAsync<SecondaryIndexTableEntity>(key.PartitionKey, key.PartitionKey, null, cancellationToken).ConfigureAwait(false);
             }
-            if (exec.Result is SecondaryIndexTableEntity secondaryIndexEntry)
+
+            if (exec.Value != null)
```

Comment:
Can this even be null given that it will throw when the entity does not exist? Also in general can we still have a way to discover something doesn't exists without having to rely on exceptions somehow?